src/map/mob.c

```diff
@@ -4719,6 +4712,85 @@ static void mob_drop_ratio_adjust(void){
 }
 
 /**
+ * Copy skill from DB to monster
+ * @param mob Monster DB entry
+ * @param skill Monster skill entries
+ **/
+static void mob_skill_db_set_single_sub(struct mob_db *mob, struct s_mob_skill *skill) {
+	uint8 i;
+
+	nullpo_retv(mob);
+	nullpo_retv(skill);
+
+	for (i = 0; mob->maxskill < MAX_MOBSKILL && i < skill->count; i++) {
+		mob->skill[mob->maxskill++] = skill->skill[i];
+	}
+
+	if (i < skill->count)
```

Comment:
alright make sense now. (u wake up on the wrong foot ?)
I couldn't test live right now so I did it in my head but ye you was right the import will cause issue about my proposition =)
just a note for thoses wondering
loop 1st cond is to ensure outofbound, 2nd is to ensure we loop tought all defined skill.
then we have the check when the skill-count is over MAX_MOBSKILL, like for say 50+2. causing us to have that cond false (mob->maxskill < MAX_MOBSKILL) so the check for skipping to act.